packages/frontend/src/pages/Users/components/UsersTable.tsx

```diff
@@ -1,121 +1,159 @@
-import { UserAvatar } from '@/components/user/UserAvatar';
-import { AllQuantifierUsers, useAdminUsers } from '@/model/users';
-import { shortenEthAddress } from '@/utils/index';
+import {
+  AllAdminUsers,
+  AllForwarderUsers,
+  AllQuantifierUsers,
+  AllUsers,
+} from '@/model/users';
 import { getUsername } from '@/utils/users';
-import { faTimesCircle } from '@fortawesome/free-solid-svg-icons';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { Dialog } from '@headlessui/react';
-import { UserDto, UserRole } from 'api/dist/user/types';
 import React from 'react';
-import { TableOptions, useTable } from 'react-table';
 import { useRecoilValue } from 'recoil';
-import PoolDeleteDialog from './DeleteDialog';
+import { UserDto, UserRole } from 'api/dist/user/types';
+import { faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import UsersTableRow from './UsersTableRow';
+import UsersTableHeader from './UsersTableHeader';
+import UsersTablePagination from './UsersTablePagination';
+
+const roleOptions = [
+  { name: 'All users', value: UserRole.USER },
+  { name: 'Admins', value: UserRole.ADMIN },
+  { name: 'Forwarders', value: UserRole.FORWARDER },
+  { name: 'Quantifiers', value: UserRole.QUANTIFIER },
+];
 
 const UsersTable = (): JSX.Element => {
+  const allAdminUsers = useRecoilValue(AllAdminUsers);
+  const allForwarderUsers = useRecoilValue(AllForwarderUsers);
   const allQuantifierUsers = useRecoilValue(AllQuantifierUsers);
-  const { removeRole } = useAdminUsers();
-
-  const deleteDialogRef = React.useRef(null);
-
-  const [isOpen, setIsOpen] = React.useState(false);
-  const [selectedQuantifier, setSelectedQuantifier] = React.useState<UserDto>();
-
-  const columns = React.useMemo(
-    () => [
-      {
-        Header: 'Id',
-        accessor: '_id',
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        Cell: (data: any): JSX.Element => {
-          return (
-            <div className="flex items-center w-full">
-              <div className="flex items-center">
-                <UserAvatar user={data.row.original} />
-              </div>
-              <div className="flex-grow p-3 whitespace-nowrap">
-                {data.row.original.accounts?.length > 0 ? (
-                  <div>{getUsername(data.row.original)}</div>
-                ) : (
-                  ''
-                )}
-                {shortenEthAddress(data.row.original.ethereumAddress)}
-              </div>
-            </div>
-          );
-        },
-      },
-    ],
-    []
+  const allUsers = useRecoilValue(AllUsers);
+  const [tableData, setTableData] = React.useState<UserDto[]>();
+  const [selectedRole, setSelectedRole] = React.useState<UserRole>(
+    UserRole.USER
   );
+  const [filter, setFilter] = React.useState<string>('');
+  const [filterType, setFilterType] = React.useState<string>('user');
```

Comment:
I think we can remove the `filterType` and conditional entirely.

Instead, always search *both* the username, and ethereumAddresses.